File: app/src/main/java/com/jarvis/app/features/TaskManager.java
package com.jarvis.app.features;

import java.util.ArrayList;
import java.util.List;

public class TaskManager {

    private final List<Task> tasks;

    public TaskManager() {
        this.tasks = new ArrayList<>();
    }

    public void addTask(String description) {
        tasks.add(new Task(description));
    }

    public void markTaskAsCompleted(int taskId) {
        if (taskId >= 0 && taskId < tasks.size()) {
            tasks.get(taskId).setCompleted(true);
        }
    }

    public List<Task> getTasks() {
        return tasks;
    }

    public static class Task {
        private static int nextId = 0;
        private final int id;
        private final String description;
        private boolean isCompleted;

        public Task(String description) {
            this.id = nextId++;
            this.description = description;
            this.isCompleted = false;
        }

        public int getId() {
            return id;
        }

        public String getDescription() {
            return description;
        }

        public boolean isCompleted() {
            return isCompleted;
        }

        public void setCompleted(boolean completed) {
            isCompleted = completed;
        }
    }
}

File: jarvis_flutter/lib/features/task_manager.dart
import 'dart:convert';
import 'dart:io';
import 'package:path_provider/path_provider.dart';
import 'package:jarvis_flutter/features/task.dart';

class TaskManager {
  static const String _fileName = 'tasks.json';
  List<Task> _tasks = [];

  Future<String> get _localPath async {
    final directory = await getApplicationDocumentsDirectory();
    return directory.path;
  }

  Future<File> get _localFile async {
    final path = await _localPath;
    return File('$path/$_fileName');
  }

  Future<void> loadTasks() async {
    try {
      final file = await _localFile;
      if (!await file.exists()) {
        _tasks = [];
        return;
      }
      final contents = await file.readAsString();
      final List<dynamic> json = jsonDecode(contents);
      _tasks = json.map((item) => Task.fromJson(item)).toList();

      if (_tasks.isNotEmpty) {
        Task.setNextId(_tasks.map((task) => task.id).reduce((a, b) => a > b ? a : b) + 1);
      }
    } catch (e) {
      _tasks = [];
    }
  }

  Future<void> _saveTasks() async {
    final file = await _localFile;
    final json = jsonEncode(_tasks.map((task) => task.toJson()).toList());
    await file.writeAsString(json);
  }

  void addTask(String description) {
    _tasks.add(Task(description));
    _saveTasks();
  }

  void deleteTask(int taskId) {
    _tasks.removeWhere((task) => task.id == taskId);
    _saveTasks();
  }

  void markTaskAsCompleted(int taskId) {
    for (final task in _tasks) {
      if (task.id == taskId) {
        task.isCompleted = true;
        _saveTasks();
        return;
      }
    }
  }

  List<Task> getTasks() {
    return _tasks;
  }
}
